hack/observability/kube-state-metrics/crd-config.yaml

```diff
@@ -264,24 +280,66 @@ spec:
       each:
         info:
           labelsFromPath:
+            container_runtime_version:
+            - status
+            - nodeInfo
+            - containerRuntimeVersion
             failure_domain:
             - spec
             - failureDomain
-            internal_ip:
+            kernel_version:
             - status
-            - addresses
-            - "[type=InternalIP]"
-            - address
+            - nodeInfo
+            - kernelVersion
+            kubelet_version:
+            - status
+            - nodeInfo
+            - kubeletVersion
+            kube_proxy_version:
+            - status
+            - nodeInfo
+            - kubeProxyVersion
+            os_image:
+            - status
+            - nodeInfo
+            - osImage
             provider_id:
             - spec
             - providerID
             version:
             - spec
             - version
-            containerRuntimeVersion:
+        type: Info
+    - name: addresses
+      help: Address information about a machine.
+      each:
+        info:
+          labelsFromPath:
+            hostname:
```

Comment:
I think I would not expose a single metric having all ips, but a metric for each IP instead 🤔 

Should be something like: 
```
    - name: addresses
      help: Address information about a machine.
      each:
        info:
          path:
          - status
          - addresses
          labelsFromPath:
            type:
            - type
            address
            - address  
```

This way it would dynamically export a metric for each entry in the array.

Reply:
changed the configuration to your proposed solution.
Example generated metrics:

```
# HELP capi_machine_addresses Address information about a machine.
# TYPE capi_machine_addresses info
capi_machine_addresses{address="10.0.0.4",cluster_name="glippy",customresource_group="cluster.x-k8s.io",customresource_kind="Machine",customresource_version="v1beta1",name="glippy-dnxdb",namespace="org-giantswarm",type="InternalIP",uid="998fe651-704e-4d72-b447-291cc8778544"} 1
capi_machine_addresses{address="glippy-control-plane-4fg59",cluster_name="glippy",customresource_group="cluster.x-k8s.io",customresource_kind="Machine",customresource_version="v1beta1",name="glippy-dnxdb",namespace="org-giantswarm",type="InternalDNS",uid="998fe651-704e-4d72-b447-291cc8778544"} 1
```